### PR TITLE
Cfinspec 314 support configmap

### PR DESCRIPTION
**Pull request by sathish-progress**
## Description
Support K8s Configmap(s)

## Related Issue

https://chefio.atlassian.net/browse/CFINSPEC-314

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Review by dkumaras — COMMENTED** [review]
Please check and confirm

File: docs-chef-io/content/inspec/resources/k8s_config_map.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+```ruby
+describe k8s_config_maps(namespace: "NAMESPACE", name: "NAME") do
+  #...
+end
+```
+
+## Parameters
+
+`type`
+: Type of the K8s resource that is for a query.
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_config_map.md:29]: As we are setting up the type to `configmaps` internally I don't think `type` parameter is required here.

File: docs-chef-io/content/inspec/resources/k8s_config_maps.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
++++
+title = "k8s_config_maps resource"
+draft = false
+gh_repo = "inspec"
+platform = "k8s"
+
+[menu]
+[menu.inspec]
+title = "k8s_config_maps"
+identifier = "inspec/resources/k8s/K8s Config Map"
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_config_maps.md:10]:
```suggestion
identifier = "inspec/resources/k8s/K8s Config Maps"
```

File: docs-chef-io/content/inspec/resources/k8s_config_map.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+platform = "k8s"
+
+[menu]
+[menu.inspec]
+title = "k8s_config_map"
+identifier = "inspec/resources/k8s/K8s Config Map"
+parent = "inspec/resources/k8s"
++++
+
+Use the `k8s_config_map` Chef InSpec audit resource to test the configuration of specific Configuration Maps in a namespace.
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_config_map.md:14]:
```suggestion
Use the `k8s_config_map` Chef InSpec audit resource to test the configuration of specific Configuration Map in a namespace.
```

File: docs-chef-io/content/inspec/resources/k8s_config_maps.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+```ruby
+describe k8s_config_maps do
+  #...
+end
+```
+
+## Parameters
+
+`type`
+: type of the K8s resource that is for query
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_config_maps.md:30]: Same I don't think we need type parameter

File: docs-chef-io/content/inspec/resources/k8s_config_maps.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+**The config maps for default namespaces should exist**
+
+```ruby
+ describe k8s_config_maps do
+  it { should exist }
+end
+```
+
+**The config maps should for given namespace exist**
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_config_maps.md:65]:
```suggestion
**The config maps for the specified namespace should exist**
```

File: test/unit/libraries/k8s_config_map_test.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            kind: 'configmap',
+            status: {
+              phase: 'running'
+            },
+            metadata: {
+              uid: 'abcd1234',
+              name: 'configmap1',
+              namespace: 'default',
+              resourceVersion: 1234,
+              annotations: [],
```
Comment (Vasu1105) [inline test/unit/libraries/k8s_config_map_test.rb:19]: annotations and labels is not a Array if I am not wrong and may need to treat it as hash.

**Review by Vasu1105 — APPROVED** [review]
Some minor doc suggestions and Annotations and labels are k8s resource object with key value pair, as per test it considers it as array in the k8s_config_map resource. According to me that should be hash.

File: docs-chef-io/content/inspec/resources/k8s_config_map.md (hunk 3 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/k8s_config_map.md:14]: ok

File: docs-chef-io/content/inspec/resources/k8s_config_maps.md (hunk 2 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/k8s_config_maps.md:10]: ok

**Review by dkumaras — COMMENTED** [review]
Please check and confirm

File: docs-chef-io/content/inspec/resources/k8s_config_maps.md [hunk 7]
```diff
@@ -27,42 +27,42 @@ end
 ## Parameters
 
 `type`
-: type of the K8s resource that is for query
+: Type of the K8s resource that is for a query.
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_config_maps.md:30]: type also needs to be removed from here

File: docs-chef-io/content/inspec/resources/k8s_config_map.md [hunk 8]
```diff
@@ -53,15 +53,15 @@ end
 
 ## Examples
 
-### Config map for default namespaces must exist
+### Configuration map for default namespaces must exist
```
Comment (Vasu1105) [inline docs-chef-io/content/inspec/resources/k8s_config_map.md:56]: namespaces should be singular

**Pull request merged**